components/post/posts.jsx

```diff
@@ -1,43 +1,94 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useSWRInfinite } from 'swr';
 import Link from 'next/link';
 import { useUser } from '@/hooks/index';
 import fetcher from '@/lib/fetch';
+import { useCurrentUser } from '@/hooks/index';
 import { defaultProfilePicture } from '@/lib/default';
+import PostEditor from './editor';
 
 function Post({ post }) {
+  const [edit, setEdit] = useState(false);
+  const [content, setContent] = useState(post.content);
+  const [msg, setMsg] = useState('');
   const user = useUser(post.creatorId);
+  const [currUser] = useCurrentUser();
+  const makeEdit = (msg, text) => {
+    setEdit(false);
+    setMsg(msg);
+    setTimeout(() => setMsg(null), 1500);
+    if (text) setContent(text);
+  };
```

Comment:
I think this logic should be inside `PostEditor`. Howabout passing an id into it. something like:

```jsx
<PostEditor editingId="foo" />
```